File: iosApp/Sources/Components/ShuttlPillTabs.swift
import SwiftUI

/// The mock's tab switcher: a pill-shaped track holding one pill per choice, the
/// chosen one filled with the heading colour and lettered in the page colour, the
/// rest lettered in the secondary text colour.
///
/// This replaces `.pickerStyle(.segmented)` on the screens built from the mock.
/// That control is UIKit's, and it draws a bordered segment in the system font on
/// a system fill - it cannot take the type scale or the tokens, so it reads as a
/// piece of another app wherever it sits. Every pill takes an equal share of the
/// track, so labels of different lengths do not make the pills ragged.
///
/// `labels` and `selectedIndex` rather than a generic over the caller's enum: the
/// callers keep their own type and pass its labels, which keeps this file free of
/// them. Port of androidApp's `ShuttlPillTabs.kt`.
struct ShuttlPillTabs: View {
    let labels: [String]
    let selectedIndex: Int
    let onSelect: (Int) -> Void
    /// Read to VoiceOver as the group's name, since the pills replace a `Picker`
    /// that carried one.
    var accessibilityLabel: String? = nil

    var body: some View {
        HStack(spacing: Metrics.gap) {
            ForEach(Array(labels.enumerated()), id: \.offset) { index, label in
                let selected = index == selectedIndex
                Button {
                    onSelect(index)
                } label: {
                    Text(label)
                        .shuttlType(ShuttlType.labelMedium)
                        // The mock letters an unchosen pill in `#5d6462`, which is
                        // `textMuted`. Not used here, and Android made the same
                        // call: that token is documented for the hero line only
                        // and drops below its own 3:1 floor on a raised surface,
                        // which this track is.
                        .foregroundStyle(selected ? Shuttl.bg : Shuttl.textSecondary)
                        .multilineTextAlignment(.center)
                        .frame(maxWidth: .infinity)
                        .padding(.horizontal, Metrics.tabPaddingH)
                        .padding(.vertical, Metrics.tabPaddingV)
                        .background(selected ? Shuttl.textHeading : Shuttl.bgSecondary)
                        .clipShape(Capsule())
                        .contentShape(Capsule())
                }
                .buttonStyle(.plain)
                .accessibilityAddTraits(selected ? [.isButton, .isSelected] : .isButton)
            }
        }
        .padding(Metrics.trackPadding)
        .frame(maxWidth: .infinity)
        .background(Shuttl.bgSecondary)
        .clipShape(Capsule())
        .accessibilityElement(children: .contain)
        .accessibilityLabel(accessibilityLabel ?? "")
    }

    /// Mirrors androidApp's `ShuttlPillTabs.kt` number for number.
    private enum Metrics {
        static let trackPadding: CGFloat = 4
        static let gap: CGFloat = 4
        static let tabPaddingH: CGFloat = 8
        static let tabPaddingV: CGFloat = 11
    }
}

File: androidApp/src/main/java/com/badmintontracker/android/ui/components/ShuttlPillTabs.kt
package com.badmintontracker.android.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.selection.selectable
import androidx.compose.foundation.selection.selectableGroup
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.badmintontracker.android.ui.theme.ShuttlRadius

/**
 * The mock's tab switcher: a pill-shaped track holding one pill per choice,
 * the chosen one filled with the heading colour and lettered in the page
 * colour, the rest lettered in the tertiary text colour.
 *
 * This replaces M3's SingleChoiceSegmentedButtonRow on the screens that were
 * built from the mock. That control draws a bordered, checkmarked segment,
 * which is Material's language and not this app's. Every pill takes an equal
 * share of the track, so labels of different lengths do not make the pills
 * ragged, and a long label wraps inside its pill rather than growing it.
 *
 * [labels] and [selectedIndex] rather than a generic type: the callers keep
 * their own enum and pass its labels, which keeps this file free of them.
 */
@Composable
fun ShuttlPillTabs(
    labels: List<String>,
    selectedIndex: Int,
    onSelect: (Int) -> Unit,
    modifier: Modifier = Modifier,
) {
    val shape = RoundedCornerShape(ShuttlRadius.pill)
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(shape)
            .background(MaterialTheme.colorScheme.surfaceVariant, shape)
            .padding(4.dp)
            .selectableGroup(),
        horizontalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        labels.forEachIndexed { index, label ->
            val selected = index == selectedIndex
            Box(
                modifier = Modifier
                    .weight(1f)
                    .clip(shape)
                    .background(
                        if (selected) MaterialTheme.colorScheme.onBackground else MaterialTheme.colorScheme.surfaceVariant,
                        shape,
                    )
                    .selectable(selected = selected, role = Role.Tab, onClick = { onSelect(index) })
                    .padding(horizontal = 8.dp, vertical = 11.dp),
                contentAlignment = Alignment.Center,
            ) {
                Text(
                    label,
                    style = MaterialTheme.typography.labelMedium,
                    color = if (selected) MaterialTheme.colorScheme.background else MaterialTheme.colorScheme.onSurfaceVariant,
                    textAlign = TextAlign.Center,
                )
            }
        }
    }
}
